src/main.rs

```diff
@@ -1,27 +1,59 @@
-use std::hash::Hash;
-use std::io::ErrorKind;
-use std::iter::FromIterator;
-use std::{
-    collections::HashSet,
-    fs::{self, File},
-    process::Command,
-};
-
+mod handler;
 use anyhow::{bail, Error, Result};
 use clap::{Parser, Subcommand, ValueEnum};
+use config::{Config, File, FileFormat};
 use glob::glob;
-use serde::{Deserialize, Serialize};
+use handler::*;
+use serde::Deserialize;
+use std::path::Path;
+use std::process::Command;
+use std::str;
+
+static GREENBOOT_INSTALL_PATHS: [&str; 2] = ["/usr/lib/greenboot", "/etc/greenboot"];
+static GREENBOOT_CONFIG_FILE: &str = "/etc/greenboot/greenboot.conf";
 
 #[derive(Parser)]
 #[clap(author, version, about, long_about = None)]
 #[clap(propagate_version = true)]
 struct Cli {
     #[clap(value_enum, short, long, default_value_t = LogLevel::Info)]
     log_level: LogLevel,
-
     #[clap(subcommand)]
     command: Commands,
 }
+#[derive(Debug, Deserialize)]
+struct GreenbootConfig {
+    //max reboot attempts if diagnostics fails
+    max_reboot: i32,
+}
+
+impl GreenbootConfig {
+    fn set_default() -> Self {
+        Self { max_reboot: 3 }
+    }
+
+    fn get_config() -> Self {
+        let mut config = Self::set_default();
+        let parsed =
+            Config::builder().add_source(File::new(GREENBOOT_CONFIG_FILE, FileFormat::Ini));
+        match parsed.build() {
```

Comment:
I believe it may be more Rust-like to write this as:

```suggestion
        let parsed =
            Config::builder()
              .add_source(File::new(GREENBOOT_CONFIG_FILE, FileFormat::Ini))
              .build();
        match parsed {
```

(Note: still new to Rust and leaning on ChatGPT for help here)